api/src/main/java/org/pmiops/workbench/notebooks/NotebooksApiClientTracer.java

```diff
@@ -0,0 +1,38 @@
+package org.pmiops.workbench.notebooks;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.Response;
+import io.opencensus.common.Scope;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+/** Created by brubenst on 7/22/19. */
+public class NotebooksApiClientTracer extends ApiClient {
+  private static final Tracer tracer = Tracing.getTracer();
+
+  @Override
+  public <T> ApiResponse<T> execute(Call call, Type returnType) throws ApiException {
+    Response response;
+    T data;
+    try (Scope ss =
+        tracer
+            .spanBuilderWithExplicitParent("NotebooksApiCall", tracer.getCurrentSpan())
+            .startScopedSpan()) {
+      response = call.execute();
+      data = handleResponseWithTracing(response, returnType);
+    } catch (IOException e) {
+      throw new ApiException(e);
+    }
+    return new ApiResponse<>(response.code(), response.headers().toMultimap(), data);
+  }
+
+  private <T> T handleResponseWithTracing(Response response, Type returnType) throws ApiException {
+    String targetUrl = response.request().httpUrl().encodedPath();
```

Comment:
Is this not available on the `call`? If you can just get the operation/method name, I think that would probably be sufficient for tracing purposes (e.g. put this on the parent span).

If you need the full URL, then an annotation might be more appropriate. Opening and immediately closing a span seems wrong to me.

Reply:
It is not available on `call`, turns out the request is there, but it's just protected and not accessible. I tried doing it as an annotation, but it was actually harder to get to the information on the Stackdriver Trace URL, because it wasn't visible by default, and was just sorta off to the side. This was the cleanest I could get it on the UI.

Reply:
![Screen Shot 2019-07-23 at 12 26 04 PM](https://user-images.githubusercontent.com/15351021/61729968-6a0c0780-ad46-11e9-9044-99bd5dfa348e.png)
![Screen Shot 2019-07-23 at 12 35 09 PM](https://user-images.githubusercontent.com/15351021/61729970-6a0c0780-ad46-11e9-9240-b5d9427a7b3f.png)

Reply:
The top one is with the span opened and closed, the bottom one is with the annotation. It requires opening the 'Show Events' button to see the URL. 

The other reason I like the span marker, is that that is where the response comes back, so it gives us the true length of the API call.

Reply:
OK, lets go with this for now. The placement of the response in that trace raises a lot of questions already.

How about adding some additional text here to make the event clearer, e.g. `Response received: /api/workspaces`

Reply:
Works for me. I will add response received. (I like that wording too)